doc/Cookbook/Sims_phoSimGen.md

```diff
@@ -0,0 +1,111 @@
+```
+source setup/setup_twinkles.sh current
+```
+At this point, the directories of the Twinkles repositories become available as part of the `desc` namespace, and an environment variable `$TWINKLES_DIR` is exported to the SHELL.
+The code uses this variable.
+
+### Generate the `phoSim` inputs
+
+The inputs to `phoSim` are phoSim Instance Catalogs which contains
+-  all of the meta-data characterizing the observation :
+- the astrophysical sources and their fluxes/models at the time of observation.
+
+The relevant information is represented in `Twinkles` as `desc.twinkles.TwinklesSky`. In order to create an instance of `TwinklesSky`, it requires input in the form of
+- instances of `lsst.sims.utils.ObservationMetaData`
+- The catalogDB Objects that define the properties of astrophysical sources in the form of database tables on fatboy. These are hard-coded into the class, as there is not much to choose. The exception is the classs representing SN, since we may use different SN tables. Each class of object Stars (static and variable) and Galaxies (Bugle, Disc, AGN) form their own groups in a Compound Catalog class. This is particularly important for the construction of time-delayed AGN (manifestations of strongly lensed AGNs). 
+
+#### Functionality 
+
+`desc.twinkles.TwinklesSky` implements a selection cut to remove all stars and galaxies brighter than a particular magnitude (set to 11.0 by default, but can be adjusted for
+each of these classes in terms of a g band magnitude). Finally, it has a method `writePhoSimCatalog` for serializing this information to an ascii file (phosim instance catalog) in a file format appropriate for phosim inputs. For static objects, PhoSim uses seds that are stored (with compression) in `CatSim`. For supernovae, the SEDs at each point of time are calculated on the fly from a model and written to disk. While writing out the phoSim instance catalog, the `writePhoSimCatalog` method also writes these SEDs to disk.
+
+There is an additional subtlety: If we want to be doing many pointings, it is inefficient to open too many new connections to fatboy (even though parallel connections are allowed).
+Hence, `TwinklesSky` has an attribute `availableConnections` which allows it to reuse connections made to fatboy. This is also used as an input (It can start as None) 
+
+The class constructor for `TwinklesSky` has parameters
+
+- obs_metadata : Observaton MetaData of the relevant OpSim pointing
+- brightestStar_gmag_inCat : the g band magnitude threshold for rejecting bright stars 
+- brightestGal_gmag_inCat : the g band magnitude threshold for rejecting bright Galalxies 
+- sntable : name of the supernova Table to use.
+- sn_sed_file_prefix : controls the location and filenames of the SN spectra written to disk
+- availableConnections : At the beginning, this can be None, but it is good to hand back available connections to a new instance rather than having it create the connections
+
+
```

Comment:
Are there instructions for setting up the credentials for the UW database access for whitelisted machines?  I think we need to put them here, with a link pointing to the confluence page with the instructions for non-whitelisted machines.  It can be as simple as:
* Create or modify `~/.lsst/db-auth.paf` to contain
```
database: {
    authInfo: {
        host: fatboy.phys.washington.edu
        port: 1433
        user: <user>
        password: <password>
    }
}
```
where `<user>` and `<password>` are obtained from [here](https://confluence.lsstcorp.org/pages/viewpage.action?pageId=35815565).  Set the permissions of that file by doing 
```
chmod go-rwx ~/.lsst/db-auth.paf
```
and the permissions of that directory with
```
chmod go-rwx ~/.lsst
```